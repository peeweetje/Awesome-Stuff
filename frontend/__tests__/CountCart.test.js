import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import CountCart from "../components/CountCart";

describe("<CountCart/>", () => {
  it("renders", () => {
    shallow(<CountCart count={10} />);
  });

  it("matches the snapshot", () => {
    const wrapper = shallow(<CountCart count={11} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("updates via props", () => {
    const wrapper = shallow(<CountCart count={50} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.setProps({ count: 10 });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
