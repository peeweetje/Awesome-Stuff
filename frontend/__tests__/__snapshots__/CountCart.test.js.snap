// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountCart/> matches the snapshot 1`] = `
<CountCart__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="11"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CountCart__Dot>
        11
      </CountCart__Dot>
    </CSSTransition>
  </TransitionGroup>
</CountCart__AnimationStyles>
`;

exports[`<CountCart/> updates via props 1`] = `
<CountCart__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="50"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CountCart__Dot>
        50
      </CountCart__Dot>
    </CSSTransition>
  </TransitionGroup>
</CountCart__AnimationStyles>
`;

exports[`<CountCart/> updates via props 2`] = `
<CountCart__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="10"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CountCart__Dot>
        10
      </CountCart__Dot>
    </CSSTransition>
  </TransitionGroup>
</CountCart__AnimationStyles>
`;
